### PR TITLE
[tensorboard] Let hparam render values correctly

### LINKED ISSUE #29906: [TensorBoard] add_hparams outputs undesired figures
## 🐛 Bug

The method `add_hparams` in tensorboard is amazing, however, sometimes it generates strange figures like the following figure:

截圖 2019-11-16 上午12 30 21

The second figure in the bottom row shows that there are some gaps in loss and accuracy when the x-axis is batch_size, which is unexpected. It should look like the one showing accuracy v.s. lr.

BTW, the labels of the x-axis are hard to see if they are too close to each other. It is better to show them vertically.

## To Reproduce

```
from torch.utils.tensorboard import SummaryWriter

with SummaryWriter() as w:
    for i in range(20):
        w.add_hparams(hparam_dict={'lr': 0.1 * i, 'batch_siz': 10 * i},
                      metric_dict={'hparam/accuracy': 0.9 ** i, 'hparam/loss': 0.01 * i})
    
writer.close()
```

### PR DESCRIPTION
**Pull request by lanpa**
The root cause of incorrect rendering is that numbers are treated as a string if the data type is not specified. Therefore the data is sort based on the first digit.

closes #29906 
 cc @orionr @sanekmelnikov

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit 0e7d1de3:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=0e7d1de34bccfe43b8d548e8518cc48f2014a130).

### :detective: 1 new failure recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [See CircleCI build](https://circleci.com/gh/pytorch/pytorch/4047096) pytorch_linux_xenial_py3_6_gcc5_4_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78541685) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78541685))
<details>
<summary>
<code>Automatic merge failed; fix conflicts and then commit the result.</code>
</summary>

```
CONFLICT (add/add): Merge conflict in .circleci/config.yml 
Auto-merging .circleci/config.yml 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/pytorch_build_data.py 
Auto-merging .circleci/cimodel/data/pytorch_build_data.py 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/dimensions.py 
Auto-merging .circleci/cimodel/data/dimensions.py 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/caffe2_build_data.py 
Auto-merging .circleci/cimodel/data/caffe2_build_data.py 
CONFLICT (add/add): Merge conflict in .circleci/cimodel/data/binary_build_data.py 
Auto-merging .circleci/cimodel/data/binary_build_data.py 
Automatic merge failed; fix conflicts and then commit the result. 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

**Comment by orionr** [comment]
Looks like some merge failures. Can you rebase on latest pytorch/master? Thanks.

**Comment by lanpa** [comment]
@natalialunova The bot seems not working, so I rebased to trigger the test again.
